src/Subscription.php

```diff
@@ -18,4 +18,19 @@ public function cancel()
     {
         return $this->delete();
     }
+
+    /**
+     * Previews the subscription.
+     *
+     * @return Subscription
+     */
+    public function preview(array $params = [], array $opts = [])
+    {
+        $response = $this->_client->request('post', '/subscriptions/preview', $params, $opts);
+
+        // build subscription object
+        $subscription = new self($this->_client);
+
+        return Util::convertPreviewToObject($subscription, $response['body']);
```

Comment:
The reason I'm not a fan of returning a Subscription object is because you have the ability to update and delete, which would fail without a persisted object. By returning an anonymous object you save someone from mistakenly calling these endpoints.

You can just return a JSON object here, like in the customer balance endpoint: https://github.com/Invoiced/invoiced-php/blob/master/src/Customer.php#L35

The alternative approach would be to make sure any API methods cannot be called on the preview object.

Reply:
Used generic JSON object solution. Ready for re-review